### PR TITLE
Feature/457938 gen ref number

### PR DESCRIPTION
**Pull request by johnfitzgeraldEviden**
#457938

Added create submission reference number.

### CODE REVIEW

**Review by JulianYoungEviden — APPROVED** [review]

File: src/EPR.RegulatorService.Facade.Core/Enums/CountryNames.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Facade.Core/Enums/CountryNames.cs:1]: Remove spaces please

File: src/EPR.RegulatorService.Facade.Core/Enums/RegistrationSubmissionType.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Facade.Core/Enums/RegistrationSubmissionType.cs:1]: Remove spaces please

File: src/EPR.RegulatorService.Facade.Core/Services/RegistrationSubmission/IRegistrationSubmissionService.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Facade.Core/Services/RegistrationSubmission/IRegistrationSubmissionService.cs:1]: Remove spaces please

File: src/EPR.RegulatorService.Facade.Core/Services/RegistrationSubmission/RegistrationSubmissionService.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+﻿
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Facade.Core/Services/RegistrationSubmission/RegistrationSubmissionService.cs:1]: Remove spaces please

File: src/EPR.RegulatorService.Facade.Core/TradeAntiVirus/AntivirusService.cs [hunk 5]
```diff
@@ -25,7 +25,7 @@ public async Task<HttpResponseMessage> SendFile(SubmissionType submissionType, G
                 Key = fileId,
                 Extension = Path.GetExtension(fileName),
                 FileName = Path.GetFileNameWithoutExtension(fileName),
-                Collection = GetCollectionName(submissionType.GetDisplayName()),
+                Collection = GetCollectionName(submissionType.GetDisplayName<SubmissionType>()),
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Facade.Core/TradeAntiVirus/AntivirusService.cs:28]: Did we mean to update this?

**Review by shehzad-ismail8 — CHANGES_REQUESTED** [review]
Some comments

File: src/EPR.RegulatorService.Facade.Core/TradeAntiVirus/AntivirusService.cs (hunk 5 above)
Comment (johnfitzgeraldEviden) [reply src/EPR.RegulatorService.Facade.Core/TradeAntiVirus/AntivirusService.cs:28]: Yes other wise I would be duplicating code

File: src/EPR.RegulatorService.Facade.Core/Enums/MaterialType.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿
+
+using System.ComponentModel.DataAnnotations;
+
+namespace EPR.RegulatorService.Facade.Core.Enums;
+
```
[the remark this answers is not in the document]
Comment (johnfitzgeraldEviden) [reply src/EPR.RegulatorService.Facade.Core/Enums/MaterialType.cs:6]: this is a new enum I created

**Review by shehzad-ismail8 — APPROVED** [review]

**Review by JulianYoungEviden — APPROVED** [review]

**Pull request merged**